src/Shared/HttpSys/RequestProcessing/NativeRequestContext.cs

```diff
@@ -192,19 +195,23 @@ internal CookedUrl GetCookedUrl()
 
         internal Version GetVersion()
         {
+            if (IsHttp3)
+            {
+                return HttpVersion.Version30;
+            }
             if (IsHttp2)
             {
-                return Constants.V2;
```

Comment:
Not for this PR but any reason why we can't clean up these versions from Constants?

Reply:
They're used in several other places that I didn't want to mess with at the moment.

Reply:
Yup, like I said, not right now. Sounds like it's a good beginner-friendly cleanup issue.